.github/workflows/deploy.yml

```diff
@@ -0,0 +1,53 @@
+name: Deploy
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+env:
+  NODE_VERSION: '22'
+
+jobs:
+  build:
+    name: Build production artifact
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          cache: 'npm'
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Setup Pages
+        id: pages
+        uses: actions/configure-pages@v5
+
+      - name: Install dependencies and build
+        run: |
+          npm ci
+          npm run build
+          cp ./dist/meteoschweiz-opendata-explorer/browser/index.html ./dist/meteoschweiz-opendata-explorer/browser/404.html
```

Comment:
Maybe add a comment here that this is a workaround to get single page applications working with Github pages.